source/grid/grid_tools.cc

```diff
@@ -5986,35 +6019,42 @@ namespace GridTools
                 }
         }
 
-      // convert to CRS
-      std::sort(ranks_and_indices.begin(), ranks_and_indices.end());
+      return convert_guessed_ranks_and_indices_to_crs(ranks_and_indices);
+    }
 
-      std::vector<unsigned int> ranks;
-      std::vector<unsigned int> ptr;
-      std::vector<unsigned int> indices;
 
-      unsigned int dummy_rank = numbers::invalid_unsigned_int;
+    // checks weather bounding boxes of elements intersect with bounding boxes
+    // of other rank
+    template <int spacedim>
+    std::tuple<std::vector<unsigned int>,
+               std::vector<unsigned int>,
+               std::vector<unsigned int>>
+    guess_intersection_owner(
+      const std::vector<std::vector<dealii::BoundingBox<spacedim>>>
+        &                                                 global_bboxes,
+      const dealii::RTree<dealii::BoundingBox<spacedim>> &element_bboxes,
+      const double                                        tolerance)
```

Comment:
Can you document what this function returns? Maybe the same for the function you introduced above -- a tuple of three vectors of ints is not self-explanatory.